### PR TITLE
NFT-Auction and Atomic-Swap-Auction

### PR DESCRIPTION
**Pull request by chrisnevers**
Added the ability to pay in non-network tokens in `parallelReduce` & `fork`.

```javascript
parallelReduce()
  .nonNetworkPay([ tokA, tokB, ...])
  .case(..., 
        ..., 
        ((amtA) => [ 0, [amtA, tokA], [0, tokB], ... ]), 
        ...)
```

Open to opinions on the format, but what I chose looks very similar to `withBill`, where you list the name of the non-network tokens and the order given will be the order received in payments, with network tokens being the first index of the payment tuple. If all looks good, I will write up documentation for it

### CODE REVIEW

File: examples/nft-auction/reach [hunk 1]
```diff
@@ -0,0 +1,1460 @@
+#!/bin/sh
```
Comment (jeapostrophe) [inline examples/nft-auction/reach:1]: Remove this file

File: examples/overview/build/index.main.mjs [hunk 2]
```diff
@@ -29,23 +29,23 @@ export async function Alice(ctc, interact) {
   const ctc5 = stdlib.T_Tuple([ctc1, ctc4, ctc1, ctc1]);
```
Comment (jeapostrophe) [inline examples/overview/build/index.main.mjs:29]: rm

File: examples/tut-8/build/index.main.mjs [hunk 3]
```diff
@@ -35,22 +35,22 @@ export async function Alice(ctc, interact) {
   const ctc11 = stdlib.T_Tuple([ctc0, ctc5, ctc0, ctc5, ctc2, ctc0]);
```
Comment (jeapostrophe) [inline examples/tut-8/build/index.main.mjs:35]: rm

File: hs/rsh/stdlib.rsh [hunk 4]
```diff
@@ -554,3 +554,7 @@ void intervalOpAux;
 export const intervalMul = (x, y) => intervalOpAux(x, y, imul);
 
 export const intervalDiv = (x, y) => intervalOpAux(x, y, idiv);
+
+export const maybe = (m, def, f) => fromMaybe(m, (() => def), f);
```
Comment (jeapostrophe) [inline hs/rsh/stdlib.rsh:558]: Docs

File: hs/rsh/stdlib.rsh [hunk 5]
```diff
@@ -554,3 +554,7 @@ void intervalOpAux;
...
+
+export const fromSome = (m, def) => maybe(m, def, ((x) => x));
```
Comment (jeapostrophe) [inline hs/rsh/stdlib.rsh:560]: Docs

**Review by jeapostrophe — CHANGES_REQUESTED** [review]

**Comment by jeapostrophe** [comment]
I like the format. For the name... `nonNetworkPay` seems so blegh. It is very accurate... I previously like `paySpec` or something like that. But I don't really feel strongly either way. This is fine.

File: examples/atomic-swap-auction/index.rsh [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+            (() => {
+              const mbid = highestBidder != this
+                ? declassify(interact.getBid(currentPrice))
+                : Maybe(UInt).None();
+              return ({
+                when: maybe(mbid, false, ((b) => b > currentPrice)),
+                msg : fromSome(mbid, 0)
+              });
+            }),
+            ((bid) => [ 0, [bid, tokB] ]),
```
[the remark this answers is not in the document]
Comment (chrisnevers) [reply examples/atomic-swap-auction/index.rsh:55]: It is redundant, but I personally find it easier and more consistent to have this payment format mirror a typical `pay` format, as opposed to being special-cased inside of a `parallelReduce`

[the remark this answers is not in the document]
Comment (jeapostrophe) [reply examples/atomic-swap-auction/index.rsh:55]: But is it checked?

**Comment by jeapostrophe** [comment]
Still some useless build artifacts included

**Pull request merged**